lavalink/node.py

```diff
@@ -176,39 +176,47 @@ async def _multi_try_connect(self, uri):
         backoff = ExponentialBackoff()
         attempt = 1
 
-        while self._is_shutdown is False and (self._ws is None or not self._ws.open):
+        while self._is_shutdown is False and (self._ws is None or self._ws.closed):
             try:
-                ws = self._ws = await websockets.connect(uri, extra_headers=self.headers)
-                return ws
+                ws = await self.session.ws_connect(url=uri, headers=self.headers)
             except OSError:
                 delay = backoff.delay()
                 log.debug("Failed connect attempt %s, retrying in %s", attempt, delay)
                 await asyncio.sleep(delay)
                 attempt += 1
-            except websockets.InvalidStatusCode:
-                return None
+            else:
+                self._ws = ws
+                return self._ws
 
     async def listener(self):
         """
         Listener task for receiving ops from Lavalink.
         """
-        while self._ws.open and self._is_shutdown is False:
-            try:
-                data = json.loads(await self._ws.recv())
-            except websockets.ConnectionClosed:
+        while self._is_shutdown is False and not self._ws.closed:
+            msg = await self._ws.receive()
+            if msg.type != aiohttp.WSMsgType.TEXT:
+                if msg.type is aiohttp.WSMsgType.ERROR:
+                    log.debug("Ignoring exception in listener task", exc_info=msg.data)
+                elif msg.type in (
+                    aiohttp.WSMsgType.CLOSED,
+                    aiohttp.WSMsgType.CLOSING,
+                    aiohttp.WSMsgType.CLOSE,
+                ):
+                    log.debug("Listener closing: %s", msg.extra)
```

Comment:
There are other message types, so adding a "catch rest" log message would be good. I'm still wondering whether all those (well, excluding the ones about closing) shouldn't have higher log level but I'll ask Draper about that.
```suggestion
                    log.debug("Listener closing: %s", msg.extra)
                else:
                    log.debug(
                        "WebSocket connection received unexpected message: %s:%s",
                        msg.type,
                        msg.data,
                    )
```